File: src/main/java/cn/cloudself/model/ItemDetailHtmlEntity.java
package cn.cloudself.model;

import org.hibernate.annotations.DynamicInsert;

import javax.persistence.*;

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "item_detail_html", schema = "shop")
public class ItemDetailHtmlEntity {
    private int id;
    private String version;
    private String div;

    @Id
    @Column(name = "id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Basic
    @Column(name = "version", nullable = true, length = 10)
    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    @Basic
    @Column(name = "div", nullable = true, length = 1024)
    public String getDiv() {
        return div;
    }

    public void setDiv(String div) {
        this.div = div;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ItemDetailHtmlEntity that = (ItemDetailHtmlEntity) o;

        if (id != that.id) return false;
        if (version != null ? !version.equals(that.version) : that.version != null) return false;
        if (div != null ? !div.equals(that.div) : that.div != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = id;
        result = 31 * result + (version != null ? version.hashCode() : 0);
        result = 31 * result + (div != null ? div.hashCode() : 0);
        return result;
    }
}

File: src/main/java/cn/cloudself/model/ItemDetailHtmlEntity.kt
package cn.cloudself.model

import org.hibernate.annotations.DynamicInsert

import javax.persistence.*

/**
 * @author HerbLuo
 * @version 1.0.0.d
 */
@Entity
@DynamicInsert
@Table(name = "item_detail_html", schema = "shop")
data class ItemDetailHtmlEntity(
        @get:Id
        @get:Column(name = "id", nullable = false)
        @get:GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Int = 0,

        @get:Basic
        @get:Column(name = "version", nullable = true, length = 10)
        var version: String? = null,

        @get:Basic
        @get:Column(name = "div", nullable = true, length = 1024)
        var div: String? = null
)
